opencog/reasoning/RuleEngine/rule-engine-src/pln/BackwardChainer.cc

```diff
@@ -22,67 +22,338 @@
  */
 #include "BackwardChainer.h"
 
+#include <opencog/atomutils/AtomUtils.h>
 #include <opencog/atoms/bind/BindLink.h>
 #include <opencog/guile/SchemeSmob.h>
 
-BackwardChainer::BackwardChainer(AtomSpace * as) :
-		 as_(as)
+BackwardChainer::BackwardChainer(AtomSpace * as)
+    : _as(as)
 {
-	commons_ = new PLNCommons(as_);
-	bcpm_ = new BCPatternMatch(as_);
+	_commons = new PLNCommons(_as);
+	_bcpm = new BCPatternMatch(_as);
 }
```

Comment:
BTW, a good practice is to avoid using new/delete as much as possible. No new/delete means less code and less possibilities of memory leaks. Sometimes you'd have to, but it seems not the case in this instance.